### DISCUSSION TITLE
Question concering read-only mode and howto make chanegs permanent

### DISCUSSION THREAD

**Discussion by stardusthamster** [cardonabits/haxo-hw#77] [Q&A]
Is this the correct way to disable and enable read-only mode?

```
sudo raspi-config
Performance Options -> Overlay filesystem -> Enable Overlay? NO
```
exit raspi config and reboot

after reboot do again
```
sudo raspi-config
Performance Options -> Overlay filesystem -> Enable Overlay? NO -> Make Boot Filesystem Writeable? Yes
```
exit raspi config and reboot

After reboot:  do the changes 

When finished
```
sudo raspi-config
Performance Options -> Overlay filesystem -> Enable Overlay? YES -> Make Boot Filesystem Writeable? Yes
```
exit raspi config and reboot

My question is:
in the last step: do I need to 

`Make Boot Filesystem Writeable? Yes`
or 
`Make Boot Filesystem Writeable? No`

I tried with No, but then after reboot everything was very slow.

What is the correct way?

Thank you very much!

**Comment by jcard0na** [comment]
Hi @stardusthamster,

It is safest to make the boot filesystem read-only (not writable), but it should not be a big deal if you leave it writable. 

The main problem we try to prevent by making filesystems read-only is filesystem corruption that can happen if the Raspberry Pi is turned off during a write operation.  

The Raspberry Pi SD image has two filesystems, `boot` and `root`.  The `root` filesystem is the riskier one, as that is being written to during normal operation of the Haxophone, mainly to collect logs.  By enabling the overlay system, the logs are stored in memory, so no write operations are made to the SD card.  

The `boot` filesystem, on the other hand, is not written to during normal operation.  Making it read-only is just an extra layer of safety against inadvertently making changes.  That said, I cannot think on why making it read-only would affect the performance of the Haxophone.

Hope that it helps,
